extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java

```diff
@@ -258,33 +261,46 @@ private String computeConfigPrefix(String interfaceName, ClassInfo classInfo) {
         return interfaceName;
     }
 
-    private ScopeInfo computeDefaultScope(Config config, Map.Entry<DotName, ClassInfo> entry, String configPrefix) {
-        // Initialize a default @Dependent scope as per the spec
-        ScopeInfo scopeInfo = BuiltinScope.DEPENDENT.getInfo();
+    private ScopeInfo computeDefaultScope(Capabilities capabilities, Config config, Map.Entry<DotName, ClassInfo> entry,
+            String configPrefix) {
+        ScopeInfo scopeToUse = null;
         final Optional<String> scopeConfig = config
                 .getOptionalValue(String.format(RestClientBase.REST_SCOPE_FORMAT, configPrefix), String.class);
+
         if (scopeConfig.isPresent()) {
             final DotName scope = DotName.createSimple(scopeConfig.get());
             final BuiltinScope builtinScope = BuiltinScope.from(scope);
             if (builtinScope != null) { // override default @Dependent scope with user defined one.
-                scopeInfo = builtinScope.getInfo();
-            } else {
+                scopeToUse = builtinScope.getInfo();
+            } else if (capabilities.isCapabilityPresent(Capabilities.SERVLET)) {
+                if (scope.toString().equals("javax.enterprise.context.SessionScoped")) {
```

Comment:
You should probably use some `DotName` constant or `SessionScoped.class.getName()`?

Reply:
Ok, wasn't sure about that

Will fix